src/components/ControlInput.vue

```diff
@@ -148,7 +154,22 @@ export default {
             this.$refs.input.focus();
         });
     },
+    mounted() {
+        this.inputRestore();
+    },
     methods: {
+        inputUpdate(val) {
+            if (state.getSetting('settings.buffers.shared_input')) {
+                state.ui.current_input = val;
+            } else {
+                this.buffer.current_input = val;
+            }
+        },
+        inputRestore() {
+            let currentInput = state.getSetting('settings.buffers.shared_input') ?
+                state.ui.current_input : this.buffer.current_input;
```

Comment:
Ternary should be in the form:
~~~javascript
let currentInput = state.getSetting('settings.buffers.shared_input') ?
    state.ui.current_input :
    this.buffer.current_input;
~~~